### PR TITLE
[GSTV-137] Fix product preview for presta 1.7.1

### PR DESCRIPTION
**Pull request by roma-kisel**
https://roihunter.atlassian.net/browse/GSTV-137?containerId=10001

### CODE REVIEW

**Comment by roma-kisel** [comment]
Was blocked by [this task](https://roihunter.atlassian.net/browse/GSTV-171) and now can be merged

**Comment by Sarhanpy** [comment]
@martinfryc1 Could you please take a look on this one?

File: roihunter.php [hunk 1]
```diff
@@ -444,14 +460,34 @@ private function createRhEasyProductDto() {
             $price = Product::getPriceStatic($id_product, $this->useTax());
 
             $currency = Context::getContext()->currency->iso_code;
-            $firstVariantId = (int) Db::getInstance()->getValue('select id_product_attribute from '._DB_PREFIX_.'product_attribute where id_product = '.$id_product);
+            $variantId = (int)$this->getProductVariantId($id_product, $refresh);
```
Comment (martinfryc1) [inline roihunter.php]: getProductVariantId() may be null, `(int)` should be used only for querry

File: roihunter.php [hunk 2]
```diff
@@ -509,6 +542,63 @@ private function createRhEasyCartItemDto() {
...
+        if (Tools::version_compare(_PS_VERSION_, '1.7', '<')) {
+            return $this->v16GetAddToCartVariantId();
+        } else {
+            return $this->v17GetVariantId($productId);
+        }
+    }
+
+    private function v17GetVariantId($productId) {
+        if (!Tools::getValue("group")) {
+            return 0;
```
Comment (martinfryc1) [inline roihunter.php]: return null

File: roihunter.php [hunk 3]
```diff
@@ -509,6 +542,63 @@ private function createRhEasyCartItemDto() {
...
+            // on each iteration we exclude all product variants are not related with our product using intersect
+            // and as the result of all iterations $allProductVariantsIds will be contain only one variant we want to find
+            $allProductVariantsIds = array_intersect($allProductVariantsIds, $productAttributeVariantIds);
+        }
+
+        if (count($allProductVariantsIds) == 1) {
+            // array_intersect returns a map we need to get only value
+            return array_values($allProductVariantsIds)[0];
+        } else {
+            return 0;
```
Comment (martinfryc1) [inline roihunter.php]: return null

**Comment by roma-kisel** [comment]
@martinfryc1 fixed :hammer_and_pick:

**Pull request merged**